File: AffiliateMarketing.Domain/Entities/AuditLog.cs
public class AuditLog
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public string EntityName { get; set; } = default!;
    public string Action { get; set; } = default!;
    public string? OldValues { get; set; } // JSON String
    public string? NewValues { get; set; } // JSON String
    public string PerformedBy { get; set; } = "System_User";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: affiliate_api/app/models/audit_log.py
# Source: AffiliateMarketing.Domain/Entities/AuditLog.cs
import uuid
from datetime import datetime, timezone
from sqlalchemy import String, DateTime, Text
from sqlalchemy.orm import mapped_column, Mapped
from app.core.db import Base

class AuditLog(Base):
    __tablename__ = "audit_logs"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    entity_name: Mapped[str] = mapped_column(String, nullable=False)
    action: Mapped[str] = mapped_column(String, nullable=False)
    old_values: Mapped[str | None] = mapped_column(Text, nullable=True)
    new_values: Mapped[str | None] = mapped_column(Text, nullable=True)
    performed_by: Mapped[str] = mapped_column(String, default="System_User")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
